src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,323 @@
+package org.prebid.server.analytics.reporter.greenbids;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Native;
+import com.iab.openrtb.request.Video;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.Future;
+import org.prebid.server.analytics.AnalyticsReporter;
+import org.prebid.server.analytics.model.AmpEvent;
+import org.prebid.server.analytics.model.AuctionEvent;
+import org.prebid.server.analytics.reporter.greenbids.model.AdUnit;
+import org.prebid.server.analytics.reporter.greenbids.model.CommonMessage;
+import org.prebid.server.analytics.reporter.greenbids.model.ExtBanner;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsAnalyticsProperties;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsBidder;
+import org.prebid.server.analytics.reporter.greenbids.model.GreenbidsConfig;
+import org.prebid.server.analytics.reporter.greenbids.model.HttpUtil;
+import org.prebid.server.analytics.reporter.greenbids.model.JsonUtil;
+import org.prebid.server.analytics.reporter.greenbids.model.MediaTypes;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.auction.model.BidRejectionReason;
+import org.prebid.server.auction.model.BidRejectionTracker;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.request.ExtImpPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ExtStoredRequest;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.NonBid;
+import org.prebid.server.proto.openrtb.ext.response.seatnonbid.SeatNonBid;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.UUID;
+
+public class GreenbidsAnalyticsReporter implements AnalyticsReporter {
+
+    private static final String ANALYTICS_SERVER = "https://a.greenbids.ai/";
+    public String pbuid;
+    public Double greenbidsSampling;
+    public Double exploratorySamplingSplit;
+    private final GreenbidsConfig greenbidsConfig;
+    private final JacksonMapper jacksonMapper;
+    public Boolean isSampled;
+    public String greenbidsId;
+    public String billingId;
```

Comment:
please make the fields private and final

Reply:
fixed